### ISSUE TITLE
Doesn't work with React Native 0.59.9 with orientation="horizontal"

### ISSUE THREAD

**Issue by raynor85** [darenju/react-native-flip-page#10]
It used to work with this version:
```js
"react": "16.6.1",
"react-native": "0.57.7",
```
If you create a new react native project though, it throws the following error:
```
Invariant Violation: [17,"RCTView",61,{"width":"<<NaN>>"}] is not usable as a native method argument
```

**Comment by raynor85** [comment]
Found the culprit and came up with a solution. The error is trigged by the props `width` and `secondHalfPull` within `horizontal-page.js`. If you log them, you'll see that they don't contain a numerical value.
The reason why they are not initialised is because `onLayout` in `index.js` is called after the component is mounted. A simple fix is ensuring to call it before it renders its children.

**Comment by darenju** [comment]
Thank you for reporting this, I will look into it when I have the time.

**Comment by raynor85** [comment]
To the following line of code, https://github.com/darenju/react-native-flip-page/blob/master/src/index.js#L369
just check the existence of the variable state:
```js
{!!this.state.halfWidth && !!this.state.halfHeight &&
  children.slice(from, to).map((component, index) => this.renderPage(component, from + index))}
```

**Comment by darenju** [comment]
I published the fix under `react-native-flip-page@0.4.1`. Can you please try?

**Comment by raynor85** [comment]
just tried, working fine

**Comment by darenju** [comment]
Cool. Does it fix #7 as well? You said it should be related.

**Comment by raynor85** [comment]
It does
